### PR TITLE
Support coercing char types of different lengths in hive

### PR DESCRIPTION
**Pull request by Praveen2112**
## Description
Add support  for coercing from 
- char to bigger char
- char to smaller char

## Additional context and related issues
Similar to #5530

## Release notes
 (x) Release notes are required, with the following suggested text:
```
## Hive Connector Changes
* Support coercion between `char` types of different lengths.
```

### CODE REVIEW

**Review by ebyhr — APPROVED** [review]

**Comment by Praveen2112** [comment]
@ebyhr AC

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/BaseTestHiveCoercion.java [hunk 1]
```diff
@@ -327,6 +334,12 @@ else if (getHiveVersionMajor() == 3 && isFormat.test("orc")) {
                 .put("varchar_to_smaller_varchar", Arrays.asList(
                         "ab",
                         "\uD83D\uDCB0\uD83D\uDCB0"))
+                .put("char_to_bigger_char", Arrays.asList(
```
Comment (skrzypo987) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/BaseTestHiveCoercion.java:337]: nit: if `ImmutableList` is not available then use `List.of()`. `Arrays.asList` is not cool anymore

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/BaseTestHiveCoercion.java [hunk 2]
```diff
@@ -593,6 +608,8 @@ private void assertColumnTypes(
                 .put("long_decimal_to_bounded_varchar", VARCHAR)
                 .put("varchar_to_bigger_varchar", VARCHAR)
                 .put("varchar_to_smaller_varchar", VARCHAR)
+                .put("char_to_bigger_char", CHAR)
```
Comment (skrzypo987) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/BaseTestHiveCoercion.java:611]: Is it feasible to try narrowing down to `char(0)`? Is it even possible?

**Review by skrzypo987 — APPROVED** [review]
lgtm % small comments

**Review by kokosing — APPROVED** [review]

**Comment by Praveen2112** [comment]
@skrzypo987 AC

**Pull request merged**